app/src/androidTest/java/com/google/samples/apps/sunflower/viewmodels/PlantDetailViewModelTest.kt

```diff
@@ -54,4 +66,18 @@ class PlantDetailViewModelTest {
     fun testDefaultValues() {
         assertFalse(getValue(viewModel.isPlanted))
     }
+
+    @Test
```

Comment:
Add `@UiThreadTest`, then you could avoid the crash while testing `setValue` of  `LiveData` which couldn't be called on the thread. 

This change might reduce `InstantTaskExecutorRule` & 
```
Room.inMemoryDatabaseBuilder(context, AppDatabase::class.java)
            // allowing main thread queries, just for testing
            .allowMainThreadQueries()
            .build()
``` above.

Reply:
@XinyueZ Good solution, Thx!